### PR TITLE
ISSUE-1043 delete public asset after user deletion

### PR DESCRIPTION
**Pull request by hungphan227**

### CODE REVIEW

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/probe/PublicAssetProbe.scala [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import reactor.core.scala.publisher.SMono
+
+class PublicAssetProbeModule extends AbstractModule {
+  override def configure(): Unit =
+    Multibinder.newSetBinder(binder(), classOf[GuiceProbe])
+      .addBinding()
+      .to(classOf[PublicAssetProbe])
+}
+
+class PublicAssetProbe @Inject()(publicAssetRepository: PublicAssetRepository) extends GuiceProbe {
```
Comment (quantranhong1999) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/probe/PublicAssetProbe.scala:18]: We don't need to rely on the NO REVIEW `/get method` commit to get the `PublicAssetProbe` IMO (can duplicate a bit and resolve later).

**Review by quantranhong1999 — APPROVED** [review]

File: tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAssetDeletionTaskStepTest.scala [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import org.junit.jupiter.api.{BeforeEach, Test}
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+import scala.jdk.CollectionConverters._
+
+class PublicAssetDeletionTaskStepTest {
+  val USERNAME_2: Username = Username.of("username2")
+  val PUBLIC_ASSET_URI_PREFIX = new URI("http://localhost:8080")
+  val BLOB_ID_FACTORY = new HashBlobId.Factory()
+  val BLOB_STORE = new DeDuplicationBlobStore(new MemoryBlobStoreDAO, BucketName.DEFAULT, BLOB_ID_FACTORY)
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAssetDeletionTaskStepTest.scala:20]: Can't those two be inlined in `beforeEach` as they are not eccessed by test cases?

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
Needs rebase

**Comment by vttranlina** [comment]
Needs rebase again

File: tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAssetDeletionTaskStepTest.scala [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  def shouldRemovePublicAssets(): Unit = {
+    SMono(publicAssetRepository.create(USERNAME, CREATION_REQUEST)).block()
+    SMono(publicAssetDeletionTaskStep.deleteUserData(USERNAME)).block()
+
+    assertThat(SFlux(publicAssetRepository.list(USERNAME)).collectSeq().block().asJava)
+      .isEmpty()
+  }
+
+  @Test
+  def shouldNotRemoveOtherUsersLabels(): Unit = {
```
Comment (quantranhong1999) [inline tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAssetDeletionTaskStepTest.scala:42]:
```suggestion
  def shouldNotRemoveOtherUsersAssets(): Unit = {
```

**Pull request merged**